src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -344,6 +346,23 @@ public SnapshotSummary retrieveSummaryByName(String name) {
         }
     }
 
+    @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
```

Comment:
For my future reference - When should this tag be added? Looks like it's sometimes used and sometimes not used in the SnapshotDao. Also, do we need to add any error handling around the update statement?

Reply:
The `@Transactional` annotation should be used anytime you want to treat multiple operations as a transaction. In this case I am doing multiple updates and I wouldn't want a failure halfway through to leave us in a state where half of the tables were updated and half had missing data. It provides a nice way to get some better consistency guarantees and prevent you from getting into bad states.